### ISSUE TITLE
render-engine init to non-existing folder errors post build

### ISSUE THREAD

**Issue by kjaymiller** [render-engine/render-engine#343]
## What is the problem?

When you run `render-engine init --project-folder=/non/existent/folder/` instead of creating the path you get an error during the build.

## Current Output
```sh
Owner Name: kjaymiller
Owner Email: kjaymiller@gmail.com
──────────────────────────────────────────────────── Creating Project ────────────────────────────────────────────────────
Generating App File: docs/app.py ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/kjaymiller/fake-app/.venv/lib/python3.12/site-packages/render_engine/cli.py:180 in init   │
│                                                                                                  │
│   177 │   │   project_config_path = pathlib.Path(project_folder).joinpath(project_path_name).w   │
│   178 │   │   task_generate_project_path = progress.add_task(f"Generating App File: [blue]{pro   │
│   179 │   │                                                                                      │
│ ❱ 180 │   │   project_config_path.write_text(                                                    │
│   181 │   │   │   CREATE_APP_PY_TEMPLATE.render(                                                 │
│   182 │   │   │   │   site_title=site_title,                                                     │
│   183 │   │   │   │   site_url=site_url,                                                         │
│                                                                                                  │
...
FileNotFoundError: [Errno 2] No such file or directory: 'docs/app.py'
```

## What should happen

There are two potential options

### Option 1: Fail Faster

You can use [typer to force the path to be a directory and exist.](https://typer.tiangolo.com/tutorial/parameter-types/path/#path-validations)

```
@app.command()
def init(
...
project-folder: pathilb.Path = Annotated[typer.Argument(
exists=True, dir_okay=True, file_okay=False, writable=True
)
```

### Option 2: Create if doesn't exist

Check if the folder exists and create it if it doesn't. 

In terms of manageability the option 1 is reduced maintenance for us, but I'm open to either (potentially start with 1 fix if folks complain)

**Comment by Mahhheshh** [comment]
Can I proceed with the task? I prefer option 1.

**Comment by kjaymiller** [comment]
That would be great! would you be open to implementing the fix along with #360. If not, just make sure to use the [Annotated version of Option](https://typer.tiangolo.com/tutorial/options/version/).

**Comment by Mahhheshh** [comment]
Sure, Will try to fix 360 along with this one. 
Btw I have done the setup for the project within vscode devcontainer, but I am unable to run the scripts. I am getting `ImportError: attempted relative import with no known parent package`.
